### PR TITLE
UAR-871 Implementation of payments and costs for Remove

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-871

### Change description

* Refactoring of Costs controller and tests
* New Costs service implementation for Remove
* DTO helper method changes to determine which journey is being followed
* Checks for appropriate journey when processing updates and removes
* TODO comments added where work needs to be done for Remove (validation and filing)
* Miscellaneous unit test changes

### Work checklist

- [X] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/CostsService.java [hunk 1]
```diff
@@ -15,10 +15,13 @@ public class CostsService {
 
     private static final String UPDATE_COST_DESCRIPTION = "Update Overseas Entity fee";
 
+    private static final String REMOVE_COST_DESCRIPTION = "Remove Overseas Entity fee";
+
     private static final String PAYMENT_ACCOUNT = "data-maintenance";
     private static final String RESOURCE_KIND = "overseas-entity";
     private static final String REGISTER_PRODUCT_TYPE = "register-overseas-entity";
     private static final String UPDATE_PRODUCT_TYPE = "update-overseas-entity";
+    private static final String REMOVE_PRODUCT_TYPE = "remove-overseas-entity";
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/CostsService.java:24]: Not sure who (or what) uses the product type. @markpit ?

I've defined this value but no idea if it's correct or needs communicating to someone.

Also, PG has informed me that product code is 16303 for removal. Not sure if that needs to feature in our code (web or API) somewhere?

**Comment by mwestacott** [comment]
Tested this in Docker and all seems fine. Also regression tested the Update and Registration journeys - payments still working there too.

**Comment by mwestacott** [comment]
Looking at the two code smells now (not the TODOs).

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/CostsService.java (hunk 1 above)
Comment (markpit) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/CostsService.java:24]: i think it is used in the services downstream of the api, after the JSON is sent to kafka

**Review by markpit — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/CostsService.java (hunk 1 above)
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/CostsService.java:24]: I now think this definition is ok and I can see where the 16303 code needs defining. See [this](https://companieshouse.atlassian.net/browse/UAR-890) ticket. I'll create a new PR for the `payment-reconciliation-consumer` repo to tie these things together.

**Review by dlloyd1-ch — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/CostsService.java (hunk 1 above)
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/CostsService.java:24]: PR is [here](https://github.com/companieshouse/payment-reconciliation-consumer/pull/50).

**Pull request merged**